### PR TITLE
Add abort handler

### PR DESCRIPTION
**Pull request by sudomakeinstall2**
Undeploy and remove B side stateroot and clean up ibu files

### CODE REVIEW

**Comment by sudomakeinstall2** [comment]
/hold
wait for hostPID PR to be merged

File: controllers/idle_handlers.go [hunk 1]
```diff
@@ -18,17 +18,75 @@ package controllers
...
+		if deployment.Booted {
+			return fmt.Errorf("failed abort: deployment %d in stateroot %s is booted", index, stateroot)
+		}
+		undeployIndices = append(undeployIndices, index)
+	}
+
+	// since undeploy shifts the order, undeploy in the reverse order
+	for i := len(undeployIndices) - 1; i >= 0; i-- {
+		ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy %d", i)
+		err := utils.ExecuteChrootCmd(utils.Host, ostreeUndeployCommand)
```
Comment (tsorya) [inline controllers/idle_handlers.go:52]: I think all ostree command should be part of ostreeclient

File: controllers/ibu_controller.go [hunk 2]
```diff
@@ -40,6 +40,8 @@ import (
 	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	"github.com/openshift-kni/lifecycle-agent/internal/clusterconfig"
 	"github.com/openshift-kni/lifecycle-agent/internal/extramanifest"
+
+	ostree "github.com/openshift-kni/lifecycle-agent/ibu-imager/ostreeclient"
```
Comment (tsorya) [inline controllers/ibu_controller.go:44]: i think ostreeclient is good name why to change it to ostree?

File: controllers/idle_handlers.go [hunk 3]
```diff
@@ -18,17 +18,75 @@ package controllers
...
+	"os"
+	"path/filepath"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	ostree "github.com/openshift-kni/lifecycle-agent/ibu-imager/ostreeclient"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+func cleanupUnbootedStateroot(ostreeClient *ostree.Client, stateroot string) error {
```
Comment (tsorya) [inline controllers/idle_handlers.go:31]: Actually all this can be part of ostreeclient no?

Comment (sudomakeinstall2) [reply controllers/idle_handlers.go:31]: yes, this will be moved to the new module once it is developed.

**Review by jc-rh — APPROVED** [review]

**Pull request merged**